lms/djangoapps/instructor_task/tasks_helper/module_state.py

```diff
@@ -221,6 +225,82 @@ def rescore_problem_module_state(xmodule_instance_args, module_descriptor, stude
 
 
 @outer_atomic
+def override_problem_score_module_state(xmodule_instance_args, module_descriptor, student_module, task_input):
+    '''
+    Takes an XModule descriptor and a corresponding StudentModule object, and
+    performs an override on the student's problem score.
+
+    Throws exceptions if the override is fatal and should be aborted if in a loop.
+    In particular, raises UpdateProblemModuleStateError if module fails to instantiate,
+    or if the module doesn't support overriding, or if the score used for override
+    is outside the acceptable range of scores (between 0 and the max score for the
+    problem).
+
+    Returns True if problem was successfully overriden for the given student, and False
+    if problem encountered some kind of error in overriding.
+    '''
+    # unpack the StudentModule:
+    course_id = student_module.course_id
+    student = student_module.student
+    usage_key = student_module.module_state_key
+
+    with modulestore().bulk_operations(course_id):
+        course = get_course_by_id(course_id)
+        instance = _get_module_instance_for_task(
+            course_id,
+            student,
+            module_descriptor,
+            xmodule_instance_args,
+            grade_bucket_type='rescore',
+            course=course
+        )
+
+        if instance is None:
+            # Either permissions just changed, or someone is trying to be clever
+            # and load something they shouldn't have access to.
+            msg = "No module {location} for student {student}--access denied?".format(
+                location=usage_key,
+                student=student
+            )
+            TASK_LOG.warning(msg)
+            return UPDATE_STATUS_FAILED
+
+        if not hasattr(instance, 'set_score'):
+            msg = "Scores cannot be updated for this problem type."
+            raise UpdateProblemModuleStateError(msg)
+
+        weighted_override_score = int(task_input['score'])
+        if weighted_override_score < 0 or weighted_override_score > instance.max_score():
```

Comment:
Nit: Python allows you to write these as:

```python
if not (0 <= weighted_score <= instance.max_score()):
```

Not a required change, just wanted to make sure you were aware the option existed.